drivers/input/touchscreen/atmel_mxt_ts.c

```diff
@@ -843,42 +844,47 @@ static int is_finger(const int id) {
 
 static int add_finger(const int id) {
 	int i;
+	mutex_lock(&fingerlist_lock);
 	for (i = 0; i < FINGER_ID_ARRAY_SIZE; i++) {
```

Comment:
esto ya es pal jogo bonito, pero puedes hacer un `int find_first_key_index(int id)' ya que el for se repite 3 veces.